GUI/PyQt/view3D_coronal.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import print_function
+
+from UnpatchNew import*
+import scipy.io as sio
+import matplotlib as mpl
+from collections import Counter
+
+import dicom
+import numpy as np
+import os
+import scipy.ndimage
+import dicom_numpy
+from matplotlib import pyplot as plt
+
+def load_scan(path):
+    slices = [dicom.read_file(path + '/' + s) for s in os.listdir(path)]
+    slices.sort(key=lambda x: int(x.InstanceNumber))
+    try:
+        slice_thickness = np.abs(slices[0].ImagePositionPatient[2] - slices[1].ImagePositionPatient[2])
+    except:
+        slice_thickness = np.abs(slices[0].SliceLocation - slices[1].SliceLocation)
+
+    for s in slices:
+        s.SliceThickness = slice_thickness
+    return slices
+
+def resample(image, scan, new_spacing=[1, 1, 1]):
+    # Determine current pixel spacing
+    spacing = map(float, ([scan[0].SliceThickness] + scan[0].PixelSpacing))  # scan is list
+    spacing = np.array(list(spacing))
+
+    resize_factor = spacing / new_spacing
+    new_real_shape = image.shape * resize_factor
+    new_shape = np.round(new_real_shape)
+    real_resize_factor = new_shape / image.shape
+    new_spacing = spacing / real_resize_factor
+
+    image = scipy.ndimage.interpolation.zoom(image, real_resize_factor)
+
+    return image, new_spacing
+
+data_path = "C:\\Users\hansw\Videos\\artefacts\\03\dicom_sorted\\t1_tse_tra_Kopf_Motion_0003"

```

Comment:
Please modify the path. See first comment

Reply:
This file and the next one are only functions of my 3D multiclass color coding. I will integrate it later into my GUI after I finish all the overlay functions.